winter_supplement_engine/rules.py

```diff
@@ -0,0 +1,123 @@
+from typing import TypedDict, Literal, Annotated, Optional
+from dataclasses import dataclass
+import json
+
+# the type of the messages the engine should receive
+class WinterSupplementInput(TypedDict):
+    id: str
+    numberOfChildren: Annotated[int, "Must be 0 or greater"]
+    familyComposition: Literal["single", "couple"]
+    familyUnitInPayForDecember: bool
+
+# the type of the messages the engine will send
+class WinterSupplementOutput(TypedDict):
+    id: str
+    isEligible: bool
+    baseAmount: float
+    childrenAmount: float
+    supplementAmount: float
+
+# function to validate the messages the engine receives
+def validate_input(data: dict) -> Optional[WinterSupplementInput]:
+    try:
+        # make sure all fields are present
+        required_fields = {'id', 'numberOfChildren', 'familyComposition', 'familyUnitInPayForDecember'}
+        if not all(field in data for field in required_fields):
+            missing = required_fields - set(data.keys())
+            raise ValueError(f"Missing required fields: {missing}")
+
+        # validate id
+        if not isinstance(data['id'], str):
+            raise ValueError("id must be a string")
+
+        # validate numberOfChildren
+        if not isinstance(data['numberOfChildren'], int) or data['numberOfChildren'] < 0:
+            raise ValueError("numberOfChildren must be a non-negative integer")
+
+        # validate familyComposition
+        if data['familyComposition'].lower() not in ["single", "couple"]:
```

Comment:
Allows the `familyComposition` value to be case-insensitive for better developer experience.